pydra/engine/helpers.py

```diff
@@ -13,9 +13,13 @@
 import re
 from time import strftime
 from traceback import format_exception
+import typing as ty
+import typing_inspect as tyi
```

Comment:
can we drop this and code up these inspections ourselves? or use attrs?

Reply:
we do this in other parts of the code where we gather type information.

Reply:
if you have an example, happy to see